### PR TITLE
Proxy

### PR DESCRIPTION
**Pull request by pelger**

### CODE REVIEW

File: lib/proxy.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    loadConfig(system, function(err, cfg) {
+      if (err) { return cb(err); }
+      if (cfg.enableProxy && _.find(cfg.enableProxy, function(p) { return p === target; })) {
+        pxy.generate(system, analyzed, function(err, haConfig) { 
+          if (err) { return cb(err); }
+
+          console.log('----------------------------------------------------');
+          console.log(haConfig);
+          console.log('----------------------------------------------------');
```
Comment (mcollina) [inline lib/proxy.js]: Can you please remove these?

File: lib/proxy.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+              if (!impl) { return cb(new Error(err)); }
+              if (!impl.hup) { return cb(new Error('container mismatch - no hup command')); }
+
+              if (mode === 'preview') {
+                if (analyzed.topology.containers[targetProxy.containedBy]) {
+                  host = analyzed.topology.containers[targetProxy.containedBy];
+                }
+                out.preview({cmd: 'HUP: ' + targetProxy.containerDef.name, host: host});
+              }
+              console.log('============> calling HUP');
```
Comment (mcollina) [inline lib/proxy.js]: and this one too?

File: lib/proxy.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        pxy.generate(system, analyzed, function(err, haConfig) { 
+          if (err) { return cb(err); }
+
+          console.log('----------------------------------------------------');
+          console.log(haConfig);
+          console.log('----------------------------------------------------');
+
+          targetProxies = determineTargets(analyzed);
+          async.eachSeries(targetProxies, function(targetProxy, next) {
+            debugger;
```
Comment (mcollina) [inline lib/proxy.js]: also this debugger statement.

File: lib/proxy.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+              if (mode === 'preview') {
+                if (analyzed.topology.containers[targetProxy.containedBy]) {
+                  host = analyzed.topology.containers[targetProxy.containedBy];
+                }
+                out.preview({cmd: 'HUP: ' + targetProxy.containerDef.name, host: host});
+              }
+              console.log('============> calling HUP');
+
+              debugger;
```
Comment (mcollina) [inline lib/proxy.js]: and this one too.

File: lib/api.js [hunk 5]
```diff
@@ -289,12 +309,38 @@ module.exports = function(options, loadConfig, logger, _sr, _containers, _servic
         systemRevision.repoPath = systemRoot;
         systemRevision.topology.name = target;
 
-        _deployer.deploy(user, systemId, revisionId, analyzed, systemRevision, _sr, mode, out, cb);
+        //_deployer.deploy(user, systemId, revisionId, analyzed, systemRevision, _sr, mode, out, cb);
```
Comment (mcollina) [inline lib/api.js]: why this is commented and not removed?

**Comment by pelger** [comment]
Well spotted - removing

**Pull request merged**